js/hoverzoom.js

```diff
@@ -1067,23 +1072,28 @@ var hoverZoom = {
                             srcDetails.url = src;
                             srcDetails.audioUrl = audioSrc;
                             clearTimeout(loadFullSizeImageTimeout);
-
-                            // if the action key has been pressed over an image, no delay is applied
-                            const delay = actionKeyDown || explicitCall ? 0 : (isVideoLink(srcDetails.url) ? options.displayDelayVideo : options.displayDelay);
                             
+                            // setLoadImage used to set loadHoveredImage within callack so timeout can be set outside of callback
+                            // This makes clearTimeout use the correct timeout ID
                             if (audioSrc) {
                                 chrome.runtime.sendMessage({action:'isImageBanned', url:audioSrc}, function (result) {
+                                    setLoadImage(false);
                                     if (!result) {
-                                        loadFullSizeImageTimeout = setTimeout(loadFullSizeImage, delay);
+                                        setLoadImage(true);
                                     }
                                 });
                             } else if (src) {
                                 chrome.runtime.sendMessage({action:'isImageBanned', url:src}, function (result) {
+                                    setLoadImage(false);
                                     if (!result) {
-                                        loadFullSizeImageTimeout = setTimeout(loadFullSizeImage, delay);
+                                        setLoadImage(true);
                                     }
                                 });                               
                             }
+                            
+                            // if the action key has been pressed over an image, no delay is applied
+                            const delay = actionKeyDown || explicitCall ? 0 : (isVideoLink(srcDetails.url) ? options.displayDelayVideo : options.displayDelay);
+                            if (loadHoveredImage) loadFullSizeImageTimeout = setTimeout(loadFullSizeImage, delay);
```

Comment:
I'm not quite sure I understand how this works. Isn't `chrome.runtime.sendMessage` asynchronous? I mean it returns immediately and at some later point invokes a call back function. Which means line 1096 could check `loadHoveredImage` value before the callback has opportunity to set it.

Reply:
That is true, and that is also why the bug was happening, i think. The timeout id being set by setTimeout wasn't always being cleared by clearTimeout due to it being async. I'm sure there is a better way of solving this

Reply:
I think it's better to temporarily remove that entire message sending block until a proper solution is found. Right now the value of `loadHoveredImage` would be reused from the _previous_ image (because the current image would only set it after it was already checked) which is not correct.

Reply:
If it's temporary, I could comment it out for now like
```js
/**
old code
*/
```

Reply:
I'd imagine @GrosPoulet would want to make it work at some point so commenting out seems fine to me.